src/tools/replay.cpp

```diff
@@ -136,4 +137,18 @@ void decode_file(std::string data, fs::path file) {
 	std::vector<uint8_t> bytes = base64::decode(data.c_str(), strlen(data.c_str()) * sizeof(char));
 	char* bytes_as_c_str = reinterpret_cast<char*>(bytes.data());
 	ofs.write(bytes_as_c_str, bytes.size() / sizeof(char));
+}
+
+location location_from_action(Element& action) {
+	location l;
+	std::istringstream sstr(action.GetText());
+	sstr >> l;
+	return l;
+}
+
+short short_from_action(Element& action) {
```

Comment:
Seems like a good place to use templates (if you need more than just these two).